src/bootstrap/util.rs

```diff
@@ -278,3 +278,39 @@ pub fn use_host_linker(target: &Interned<String>) -> bool {
         || target.contains("fortanix")
         || target.contains("fuchsia"))
 }
+
+pub fn clone_repository(
+    repository_url: &str,
+    target_dir: &Path,
+    specific_commit_hash: Option<&str>,
+) {
+    if target_dir.is_dir() {
+        fs::remove_dir_all(target_dir).expect("failed to remove target folder");
+    }
+    let status = Command::new("git")
+        .arg("clone")
+        .arg(repository_url)
+        .arg(target_dir.to_str().unwrap())
```

Comment:
There should be no need to unwrap here, though you might need e.g. as_os_str() or so instead of to_str(), but I think even that should be unnecessary.